consensus/broadcast/commonMessenger.go

```diff
@@ -223,3 +249,19 @@ func (cm *commonMessenger) extractMetaMiniBlocksAndTransactions(
 
 	return metaMiniBlocks, metaTransactions
 }
+
+func (cm *commonMessenger) broadcast(topic string, data []byte, pkBytes []byte) {
+	if bytes.Equal(pkBytes, cm.currentPublicKeyBytes) {
+		cm.messenger.Broadcast(topic, data)
+		return
+	}
+
+	skBytes, pid, err := cm.keysHolder.GetP2PIdentity(pkBytes)
+	if err != nil {
+		log.Error("setup error in commonMessenger.broadcast - public key is managed but does not contain p2p sign info",
+			"pk", pkBytes, "error", err)
+		return
```

Comment:
thought about this. I think part of the messages will be discarded because it will broadcast data as an observer on topics on which data needs to be sent only by validators.

Reply:
ok